app/src/pages/playground/InvocationParametersForm.tsx

```diff
@@ -16,113 +30,221 @@ const FormField = ({
-  // Get the schema for the incoming provider and model combination.
-  const schema = getInvocationParametersSchema({
-    modelProvider: provider,
-    modelName: modelName || "default",
-  });
+  const { model } = instance;
+  const updateInstanceModelInvocationParameters = usePlaygroundContext(
+    (state) => state.updateInstanceModelInvocationParameters
+  );
+  const filterInstanceModelInvocationParameters = usePlaygroundContext(
+    (state) => state.filterInstanceModelInvocationParameters
+  );
+  const { modelInvocationParameters } =
+    useLazyLoadQuery<InvocationParametersFormQuery>(
+      graphql`
+        query InvocationParametersFormQuery($input: ModelsInput!) {
+          modelInvocationParameters(input: $input) {
+            __typename
+            ... on InvocationParameterBase {
+              invocationName
+              label
+              required
+              canonicalName
+            }
+            ... on BoundedFloatInvocationParameter {
+              minValue
+              maxValue
+              invocationInputField
+              # defaultValueFloat: defaultValue
+            }
+            ... on IntInvocationParameter {
+              invocationInputField
+              # defaultValueInt: defaultValue
+            }
+            ... on StringInvocationParameter {
+              invocationInputField
+              # defaultValueString: defaultValue
+            }
+            ... on StringListInvocationParameter {
+              invocationInputField
+              # defaultValueStringList: defaultValue
+            }
+            ... on BooleanInvocationParameter {
+              invocationInputField
+              # defaultValueBool: defaultValue
+            }
+          }
+        }
+      `,
+      { input: { providerKey: model.provider, modelName: model.modelName } }
+    );
+
+  useEffect(() => {
+    // filter invocation parameters to only include those that are supported by the model
+    if (modelInvocationParameters) {
+      filterInstanceModelInvocationParameters({
+        instanceId: instance.id,
+        modelSupportedInvocationParameters:
+          modelInvocationParameters as Mutable<
+            typeof modelInvocationParameters
+          >,
+      });
+    }
+  }, [
+    filterInstanceModelInvocationParameters,
+    instance.id,
+    modelInvocationParameters,
+  ]);
+
+  const onChange = useCallback(
+    (
+      field: InvocationParameter,
+      value: string | number | string[] | boolean | undefined
+    ) => {
+      const existingParameter = instance.model.invocationParameters.find(
+        (p) => p.invocationName === field.invocationName
+      );
+
+      if (existingParameter) {
+        const input = makeInvocationParameterInput(field, value);
+        if (input) {
+          updateInstanceModelInvocationParameters({
+            instanceId: instance.id,
+            invocationParameters: instance.model.invocationParameters.map(
```

Comment:
remind me again why this got changed to a list

Reply:
ahh gotcha so the api returns a list?

Reply:
yeah, the UI doesn't need to care about models or invocation parameters anymore, only how to handle *types* of invocation parameters. The process is now:

- pick a provider/model
- ask gql for the supported invocation parameters, get handed a list of *types*
- UI renders the types, and passes them all back

Reply:
This way new models can be added with no UI changes at all